### ISSUE TITLE
[mig agent] sandbox javascript interpreter to perform pre/post processing of operations (Bugzilla #1155981)

### ISSUE THREAD

**Issue by jvehent** [mozilla/mig#119]
Migrated from https://bugzilla.mozilla.org/show_bug.cgi?id=1155981
Assigned to: Julien Vehent [:ulfr]

On 2015-04-18 08:30:18 -0700, Julien Vehent [:ulfr] wrote:

> When an agent executes N operations, each operation is ran in parallel against its own instantiation of a module. Results are aggregated into a JSON array and sent back to the scheduler. Modules don't know about each other and have no way to interact.
> 
> The ability to chain the work of modules would open new forensics options. The proposed workflow is to chain operations like unix commands are chained with pipes: the output of a module could become the input of another module. 
> To achieve this, we need a data transformation layer that convert the output of operation 1 into an input for operation 2. That layer could be performed by a lightweight javascript interpreter. Instead of running in parallel, operations would run in sequence, following the ordering of the operations array, with javascript transformation performed in between each operation.
> 
> The javascript interpreter must run into a sandbox. It must not be allowed to perform IOs. The agent must control the stdin and stdout of a javascript sandbox.

**Comment by jvehent** [comment]
It looks like the `scribe` module may be a better of way of achieving the same result. @ameihm0912 : could you review this issue and wontfix it if you think it's unnecessary?

**Comment by ameihm0912** [comment]
Closing this old issue as there hasn't been anything done here, and the approach itself may deserve being revisited anyway.
